core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterPersistenceEngine.scala

```diff
@@ -94,7 +94,7 @@ private[spark] class ZookeeperMesosClusterPersistenceEngine(
     conf: SparkConf)
   extends MesosClusterPersistenceEngine with Logging {
   private val WORKING_DIR =
-    conf.get("spark.deploy.zookeeper.dir", "/spark_mesos_dispatcher") + "/" + baseDir
+    conf.get("spark.mesos.deploy.zookeeper.dir", "/spark_mesos_dispatcher") + "/" + baseDir
```

Comment:
Up to you, but does it make sense to fall back to the original property as a default to preserve compatibility? Does that make sense for all these Mesos-specific override properties?